libnetwork/testutils/context.go

```diff
@@ -0,0 +1,41 @@
+package testutils
+
+import "testing"
+
+// Logger is used to log non-fatal messages during tests.
+type Logger interface {
+	Logf(format string, args ...any)
+}
+
+var _ Logger = (*testing.T)(nil)
+
+// SetupTestOSContext joins the current goroutine to a new network namespace,
+// and returns its associated teardown function.
+//
+// Example usage:
+//
+//	defer SetupTestOSContext(t)()
+func SetupTestOSContext(t *testing.T) func() {
+	c := SetupTestOSContextEx(t)
+	return func() { c.Cleanup(t) }
```

Comment:
The consequences of forgetting to call the cleanup function are fairly minor. It will leak two open file descriptors and a network namespace for the life of the test process, and cause an OS thread to be terminated and replaced. `defer`s are executed before `t.Cleanup()`s so your suggestion would work like a charm. That's a good one to keep in our back pocket if we find it necessary to make cleanup mandatory.

Reply:
yeah, it was just an idea 😄 

was half-way through your commits, and thought "let me post these already"; will finish up reviewing later (so far looks great!)